src/network/connection.js

```diff
@@ -304,43 +304,43 @@ module.exports = class Connection {
 
     this.buffer = Buffer.concat([this.buffer, rawData])
 
-    // Not enough bytes to read the expected response size, keep buffering
-    if (Buffer.byteLength(this.buffer) <= Decoder.int32Size()) {
-      return
-    }
-
-    const data = Buffer.from(this.buffer)
-    const decoder = new Decoder(data)
-    const expectedResponseSize = decoder.readInt32()
+    // Process data if there are enough bytes to read the expected response size,
+    // otherwise keep buffering
+    while (Buffer.byteLength(this.buffer) > Decoder.int32Size()) {
+      const data = Buffer.from(this.buffer)
+      const decoder = new Decoder(data)
+      const expectedResponseSize = decoder.readInt32()
+
+      if (!decoder.canReadBytes(expectedResponseSize)) {
+        return
+      }
 
-    if (!decoder.canReadBytes(expectedResponseSize)) {
-      return
-    }
+      const response = new Decoder(decoder.readBytes(expectedResponseSize))
+      // Reset the buffer as the rest of the bytes
+      this.buffer = decoder.readAll()
 
-    // The full payload is loaded, erase the temporary buffer
-    this.buffer = Buffer.alloc(0)
+      if (this.authHandlers) {
+        return this.authHandlers.onSuccess(data.slice(0, this.offset + byteLength))
```

Comment:
`byteLength` isn't defined, this is breaking the tests

Reply:
good catch! copy-paste bug...